src/mad_spark_alt/unified_cli.py

```
@@ -744,17 +782,81 @@ async def _run_qadi_analysis(
+    ollama_provider = None
+    try:
+        import aiohttp
+        ollama_provider = OllamaProvider()
+        # Actually verify Ollama server is reachable
+        session = await ollama_provider._get_session()
+        async with session.get(
+            f"{ollama_provider.base_url}/api/tags",
+            timeout=aiohttp.ClientTimeout(total=2)
+        ) as resp:
+            if resp.status != 200:
+                raise ConnectionError("Ollama server not responding")
```

Ruff (0.14.4)

Abstract `raise` to an inner function

(TRY301)